File: JK/14. Longest Common Prefix.py

def longestCommonPrefix(strs):
    prefix = strs[0]
    for i in range(1,len(strs)):
        if len(prefix) > len(strs[i]):
            prefix = strs[i]

    for i in range(len(strs)):
        while len(prefix) > 0:
            if strs[i][:len(prefix)]== prefix:
                break
            else:
                prefix = prefix[:len(prefix)-1]


    return prefix

File: JK/14. Longest Common Prefix.c
#include <stdio.h>
#include <string.h>


char* longestCommonPrefix(char **strs, int strsSize) {

    if (strsSize == 0) return "";

    char *prefix = strs[0];

    for (int i = 1; i < strsSize; i++) {
        if (strlen(prefix) > strlen(strs[i])) {
            prefix = strs[i];
        }
    }

    for (int i = 0; i < strsSize; i++) {
        while (strlen(prefix) > 0) {
            if (strncmp(strs[i], prefix, strlen(prefix)) == 0) {
                break;
            } else {
                prefix[strlen(prefix) - 1] = '\0';
            }
        }
    }

    return prefix;
}




int main() {
    // Example usage
    char *strs[] = {"flower", "flow", "flight"};
    int strsSize = 3;

    char *result = longestCommonPrefix(strs, strsSize);
    printf("Longest common prefix: %s\n", result);

    return 0;
}
